storage/writer.go

```diff
@@ -113,6 +113,13 @@ type Writer struct {
 	// yet available for general use.
 	Append bool
 
+	// FinalizeOnClose indicates whether the writer should finalize an object when
+	// closing the write stream. This only applies to writers where Append is
+	// true, since append semantics allow a prefix of the object to be durable and
+	// readable. For other writers, the correct way to abandon an upload is to
+	// cancel its context.
+	FinalizeOnClose bool
```

Comment:
So IIUC this would mean that Appendable objects are never finalized by default. Does this square with how we expect appends to typically be used?

I was originally thinking we should make the param `CloseWithoutFinalize` to make finalization the default and ensure that people explicitly opt in to leaving open objects, but either way would be fine ultimately.

Reply:
Ah, I see that you make it true by default in NewWriter. I am not a fan of this because people expect default value of false for a bool in Go. If we want to default to finalization, let's do what I suggested as far as the param name.